app/src/main/java/com/hedvig/app/feature/offer/ui/OfferActivity.kt

```diff
@@ -127,20 +132,27 @@ class OfferActivity : BaseActivity(R.layout.activity_offer) {
                     ConcatSpanSizeLookup(gridLayoutManager.spanCount) { concatAdapter.adapters }
             }
 
-            model.viewState.observe(this@OfferActivity) { viewState ->
-                when (viewState) {
-                    OfferViewModel.ViewState.HasContracts -> startLoggedInActivity()
-                    is OfferViewModel.ViewState.OfferItems -> {
-                        topOfferAdapter.submitList(viewState.topOfferItems)
-                        perilsAdapter.submitList(viewState.perils)
-                        insurableLimitsAdapter.submitList(viewState.insurableLimitsItems)
-                        documentAdapter.submitList(viewState.documents)
-                        bottomOfferAdapter.submitList(viewState.bottomOfferItems)
+            lifecycleScope.launchWhenResumed {
+                model.viewState.collect { viewState ->
```

Comment:
In order to 'mimic' the desirable `Lifecycle`-behaviour of `LiveData` (collect when > `onResume`, pause in `onPause`, restart collection in `onResume`, etc), we should use `flowWithLifecycle` to collect this flow.

```suggestion
                model.viewState.flowWithLifecycle(this, Lifecycle.State.STARTED).collect { viewState ->
```

Here's an article on the subject: https://medium.com/androiddevelopers/a-safer-way-to-collect-flows-from-android-uis-23080b1f8bda 🙂

Reply:
Nice